client/asset/eth/rpcclient.go

```diff
@@ -256,6 +261,64 @@ func (c *rpcclient) initiate(txOpts *bind.TransactOpts, netID int64, refundTimes
 	return c.es.Initiate(txOpts, big.NewInt(refundTimestamp), secretHash, *participant)
 }
 
+// initiateGas checks the amount of gas that is used for a call to the initiate function.
+func (c *rpcclient) initiateGas(ctx context.Context, contractAddress *common.Address) (uint64, error) {
+	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
+	if err != nil {
+		return 0, err
+	}
+	refundTimestamp := time.Now().Unix()
+	var secret [32]byte
+	copy(secret[:], encode.RandomBytes(32))
+	secretHash := sha256.Sum256(secret[:])
+	// just an arbitrary address
+	participant := common.HexToAddress("345853e21b1d475582E71cC269124eD5e2dD3422")
+	data, err := parsedAbi.Pack("initiate", big.NewInt(refundTimestamp), secretHash, participant)
+	if err != nil {
+		return 0, nil
+	}
+
+	msg := ethereum.CallMsg{
+		From:  participant,
+		To:    contractAddress,
+		Gas:   0,
+		Value: big.NewInt(100000000),
```

Comment:
I think this is one less power of 10 wei than one gwei, so less than a value we can actually send, as we are rounding everything to gwei. If so could you add a 0?